gcp-client/src/main/java/com/google/graphite/platforms/plugin/client/ComputeClient.java

```diff
@@ -0,0 +1,541 @@
+      }
+    } catch (InterruptedException ie) {
+      // catching InterruptedException here because calling function also can throw
+      // InterruptedException from trying to terminate node
+      LOGGER.log(Level.WARNING, "Error in creating snapshot.", ie);
+      throw ie;
+    } catch (IOException ioe) {
+      LOGGER.log(Level.WARNING, "Interruption in creating snapshot.", ioe);
+      throw ioe;
+    }
+  }
+
+  /**
+   * Given a disk's name, create a snapshot for said disk.
+   *
+   * @param projectId Google cloud project id.
+   * @param zone Zone of disk.
+   * @param diskId Name of disk to create a snapshot for.
+   * @throws IOException If an error occured in snapshot creation.
+   * @throws InterruptedException If snapshot creation is interrupted.
+   */
+  public void createSnapshotForDisk(String projectId, String zone, String diskId)
+      throws IOException, InterruptedException {
+    Snapshot snapshot = new Snapshot();
+    snapshot.setName(diskId);
+
+    Operation op = compute.disks().createSnapshot(projectId, zone, diskId, snapshot).execute();
+    // poll for result
+    waitForOperationCompletion(
+        projectId, op.getName(), op.getZone(), SNAPSHOT_TIMEOUT_MILLISECONDS);
+  }
+
+  /**
+   * Deletes persistent disk snapshot. Does not block.
+   *
+   * @param projectId Google cloud project id.
+   * @param snapshotName Name of the snapshot to be deleted.
+   * @throws IOException If an error occurred in deleting the snapshot.
+   */
+  public void deleteSnapshot(String projectId, String snapshotName) throws IOException {
+    compute.snapshots().delete(projectId, snapshotName).execute();
+  }
+
+  /**
+   * Returns snapshot with name snapshotName
+   *
+   * @param projectId Google cloud project id.
+   * @param snapshotName Name of the snapshot to get.
+   * @return Snapshot object with given snapshotName. Null if not found.
+   * @throws IOException If an error occurred in retrieving the snapshot.
+   */
+  public Snapshot getSnapshot(String projectId, String snapshotName) throws IOException {
+    SnapshotList response;
+    Compute.Snapshots.List request = compute.snapshots().list(projectId);
+
+    do {
+      response = request.execute();
+      if (response.getItems() == null) {
+        continue;
+      }
+      for (Snapshot snapshot : response.getItems()) {
+        if (snapshotName.equals(snapshot.getName())) return snapshot;
+      }
+      request.setPageToken(response.getNextPageToken());
+    } while (response.getNextPageToken() != null);
+
+    return null;
+  }
+
+  /**
+   * Appends metadata to an instance. Any metadata items with existing keys will be overwritten.
+   * Otherwise, metadata is preserved. This method blocks until the operation completes.
+   *
+   * @param projectId
+   * @param zone
+   * @param instanceId
+   * @param items
+   * @throws IOException
+   * @throws InterruptedException
+   */
+  public Operation.Error appendInstanceMetadata(
```

Comment:
I'm concerned about mixing error returns and exceptions. Typically Java prefers the latter so you don't forget to check them.

Reply:
I've opted to have all of these return Operation instead and throw a wrapper exception to avoid mixing these up.